system-tests/src/test/java/org/eclipse/edc/samples/util/HttpRequestLoggerContainer.java

```diff
@@ -14,28 +14,34 @@
 
 package org.eclipse.edc.samples.util;
 
+import org.eclipse.edc.util.io.Ports;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.output.ToStringConsumer;
 import org.testcontainers.images.builder.ImageFromDockerfile;
 
-import java.util.List;
-
 import static org.eclipse.edc.samples.common.FileTransferCommon.getFileFromRelativePath;
 
 public class HttpRequestLoggerContainer extends GenericContainer<HttpRequestLoggerContainer> {
 
     private static final String HTTP_REQUEST_LOGGER_DOCKERFILE_PATH = "util/http-request-logger/Dockerfile";
     private static final ImageFromDockerfile IMAGE_FROM_DOCKERFILE = new ImageFromDockerfile()
             .withDockerfile(getFileFromRelativePath(HTTP_REQUEST_LOGGER_DOCKERFILE_PATH).toPath());
-    private static final String PORT_BINDING = "4000:4000";
+    private final ToStringConsumer toStringConsumer;
+    private final int port;
 
     public HttpRequestLoggerContainer() {
         super(IMAGE_FROM_DOCKERFILE);
-        this.setPortBindings(List.of(PORT_BINDING));
+        this.toStringConsumer =  new HttpRequestLoggerConsumer();
+        this.port =  Ports.getFreePort();
+        this.withLogConsumer(toStringConsumer)
+                .addFixedExposedPort(this.port, 4000);
```

Comment:
The testcontainers docs report:
> Note that this method is protected scope to discourage use, as clashes or instability are more likely when using fixed port mappings

I would suggest to use the `withExposedPort(4000)` and in the `getPort()` method return the value of `getFirstMappedPort`, do testcontainers will take care to instantiate a random port on startup

Reply:
I'll do that now

Reply:
Fixed https://github.com/eclipse-edc/Samples/pull/364/commits/6154fe28f2eea3b9dcd8166031806c0daeefeba7